--- a/contrib/relic/CMakeLists.txt
+++ b/contrib/relic/CMakeLists.txt
@@ -369,7 +369,7 @@
 	endif(GMP_FOUND)
 endif(ARITH STREQUAL "gmp" OR ARITH STREQUAL "gmp-sec")

-set(CMAKE_C_FLAGS "-pipe -std=c99 ${AFLAGS} ${WFLAGS} ${DFLAGS} ${PFLAGS} ${CFLAGS}")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -std=c99 ${AFLAGS} ${WFLAGS} ${DFLAGS} ${PFLAGS} ${CFLAGS}")
 set(CMAKE_EXE_LINKER_FLAGS ${LFLAGS})

 message(STATUS "Compiler flags: ${CMAKE_C_FLAGS}")
